ansible/roles/docker/tasks/main.yml

```diff
@@ -24,6 +24,8 @@
     dest: /etc/systemd/system/docker.service.d
     state: directory
 
+# Required by the docker_image ansible module
 - name: install docker-py
   pip:
     name: docker-py
+  when: kubernetes_enable_cached_images | bool
```

Comment:
Are we sure that the docker role has access to this variable?  It does not appear to.

As this is not really something specific to the "docker" role, and more a function of the "kubernetes" role, it probably belongs over there. I also think the conditional needs to look like `kubernetes_enable_cached_images|bool and kubernetes_cached_images|len > 0` or just plain old `kubernetes_cached_images|len > 0`

Reply:
Good call. Seems like moving the installation of docker-py to the kubernetes role would make everything cleaner. Thanks @craigtracey!